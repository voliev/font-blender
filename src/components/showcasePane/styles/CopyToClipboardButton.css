.CopyToClipboardButton .icon {
  display: flex;
  margin-right: var(--space-xs);
}
