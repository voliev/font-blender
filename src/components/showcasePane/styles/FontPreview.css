.FontPreview {
  margin: 0 auto;
  min-height: calc(100vh - 4rem);
  padding: var(--space-lg) var(--space-md) var(--space-xxl) var(--space-md);
  width: 50rem;
}

.FontPreview.error,
.FontPreview.fetching {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.FontPreview.error .msg {
  display: flex;
  flex-direction: column;
  padding-left: var(--space-xl);
  position: relative;
}

.FontPreview.error .msg .text:first-of-type {
  font-weight: 700;
  margin: var(--space-xs) 0 var(--space-xxs) 0;
}

.FontPreview.error .msg .text:last-of-type {
  margin: var(--space-xxs) 0 var(--space-lg) 0;
}

.FontPreview.error .msg svg {
  position: absolute;
  top: 0;
  left: 0;
}

.FontPreview.error .btn {
  align-self: stretch;
}
