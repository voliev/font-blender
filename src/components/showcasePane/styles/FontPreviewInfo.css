.FontPreviewInfo .heading {
  border-bottom: 1px solid var(--light-grey);
  padding-bottom: var(--space-md);
}

.FontPreviewInfo .category-name,
.FontPreviewInfo .charsets-list {
  color: var(--dark-grey);
}

.FontPreviewInfo .charsets-list {
  padding-left: var(--space-md);
}
