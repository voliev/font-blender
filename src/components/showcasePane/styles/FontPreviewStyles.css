.FontPreviewStyles .heading {
  border-bottom: 1px solid var(--light-grey);
  padding-bottom: var(--space-md);
}

.FontPreviewStyles .font-size-controls {
  align-items: center;
  display: flex;
}
