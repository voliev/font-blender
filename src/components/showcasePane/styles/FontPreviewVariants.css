.FontPreviewVariants .variant {
  align-items: center;
  display: flex;
}

.FontPreviewVariants .variant-value {
  flex: 0 0 7rem;
  margin-right: var(--space-md);
}

.FontPreviewVariants .variant-example {
  flex: 0 auto auto;
  margin: 0;
}
