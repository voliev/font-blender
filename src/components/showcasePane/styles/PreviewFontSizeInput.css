.PreviewFontSizeInput {
  align-items: center;
  display: flex;
}

.PreviewFontSizeInput.label,
.PreviewFontSizeInput.input {
  margin-right: var(--space-md);
}

.PreviewFontSizeInput.label {
  flex: 0 0 7rem;
}

.PreviewFontSizeInput.input {
  flex: auto;
}
