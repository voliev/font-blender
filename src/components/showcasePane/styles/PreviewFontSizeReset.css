.PreviewFontSizeReset {
  height: 2rem;
  padding: var(--space-xs) var(--space-md);
}
