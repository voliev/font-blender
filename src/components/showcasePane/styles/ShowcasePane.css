.ShowcasePane {
  min-height: calc(100vh - 4rem);
  overflow-y: scroll;
  position: relative;
  -webkit-overflow-scrolling: touch;
  width: calc(100vw - 2 * 22rem);
}

.ShowcasePane .error-container {
  align-items: center;
  display: flex;
  height: 100%;
  justify-content: center;
}

.ShowcasePane .error-msg {
  font-size: var(--font-md);
  font-weight: 700;
  text-align: center;
  width: 100%;
}
