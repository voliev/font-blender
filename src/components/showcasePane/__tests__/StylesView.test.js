import React from "react";
import { mount, shallow } from "enzyme";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { Light as SyntaxHighlighter } from "react-syntax-highlighter";
import css from "react-syntax-highlighter/dist/languages/hljs/css";
import docco from "react-syntax-highlighter/dist/styles/hljs/docco";
import StylesViewConnected, { StylesView } from "../StylesView";
import template from "../../../template/styles";
import state from "../../../fixtures/state";
import fonts from "../../../fixtures/fonts";
import styles from "../../../fixtures/styles";

const setup = (renderFn = shallow, propsOverrides = {}, connected = false) => {
  const props = {
    fonts,
    styles,
    ...propsOverrides
  };

  if (connected) {
    const mockStore = configureStore();
    const store = mockStore(state);

    return {
      store,
      wrapper: renderFn(
        <Provider store={store}>
          <StylesViewConnected />
        </Provider>
      )
    };
  }

  return {
    ...props,
    wrapper: renderFn(<StylesView {...props} />)
  };
};

it("should render without crashing", () => {
  const { wrapper } = setup();

  expect(wrapper).toMatchSnapshot();
});

it("should receive corresponding props", () => {
  const { wrapper } = setup(mount, undefined, true);

  expect(typeof wrapper.find(StylesView).prop("styles")).toBe("object");
  expect(wrapper.find(StylesView).prop("styles")).toEqual(styles);

  expect(typeof wrapper.find(StylesView).prop("fonts")).toBe("object");
  expect(wrapper.find(StylesView).prop("fonts")).toEqual(fonts);
});

it("should not update component", () => {
  const { wrapper } = setup();
  const shouldComponentUpdate = jest.spyOn(
    StylesView.prototype,
    "shouldComponentUpdate"
  );

  wrapper.instance().shouldComponentUpdate();

  expect(shouldComponentUpdate).toHaveReturnedWith(false);
});

it("should pass on corresponding props to SyntaxHighlighter component", () => {
  const { wrapper } = setup();

  expect(wrapper.find(SyntaxHighlighter).prop("language")).toBe("css");
  expect(wrapper.find(SyntaxHighlighter).prop("style")).toBe(docco);
  expect(wrapper.find(SyntaxHighlighter).prop("customStyle")).toEqual({
    whiteSpace: "pre-wrap",
    borderRadius: "6px",
    padding: "2rem"
  });
  expect(wrapper.find(SyntaxHighlighter).prop("id")).toBe("code-snippet");
  expect(wrapper.find(SyntaxHighlighter).prop("children")).toEqual(
    template(styles, fonts)
  );
});

it("should register language for syntax highlighting", () => {
  const registerLanguage = jest.spyOn(SyntaxHighlighter, "registerLanguage");

  setup();

  expect(registerLanguage).toHaveBeenCalledTimes(1);
  expect(registerLanguage).toHaveBeenCalledWith("css", css);
});
