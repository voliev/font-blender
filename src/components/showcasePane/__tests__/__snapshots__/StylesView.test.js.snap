// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<div
  className="StylesView"
>
  <SyntaxHighlighter
    customStyle={
      Object {
        "borderRadius": "6px",
        "padding": "2rem",
        "whiteSpace": "pre-wrap",
      }
    }
    id="code-snippet"
    language="css"
    style={
      Object {
        "hljs": Object {
          "background": "#f8f8ff",
          "color": "#000",
          "display": "block",
          "overflowX": "auto",
          "padding": "0.5em",
        },
        "hljs-addition": Object {
          "background": "#dfd",
        },
        "hljs-attribute": Object {
          "color": "#000080",
          "fontWeight": "normal",
        },
        "hljs-built_in": Object {
          "color": "#0086b3",
        },
        "hljs-builtin-name": Object {
          "color": "#0086b3",
        },
        "hljs-bullet": Object {
          "color": "#990073",
        },
        "hljs-comment": Object {
          "color": "#408080",
          "fontStyle": "italic",
        },
        "hljs-deletion": Object {
          "background": "#fdd",
        },
        "hljs-doctag": Object {
          "color": "#219161",
        },
        "hljs-emphasis": Object {
          "fontStyle": "italic",
        },
        "hljs-keyword": Object {
          "color": "#954121",
        },
        "hljs-link": Object {
          "color": "#b68",
        },
        "hljs-literal": Object {
          "color": "#954121",
        },
        "hljs-meta": Object {
          "color": "#999",
          "fontWeight": "bold",
        },
        "hljs-name": Object {
          "color": "#000080",
          "fontWeight": "normal",
        },
        "hljs-number": Object {
          "color": "#40a070",
        },
        "hljs-params": Object {
          "color": "#00f",
        },
        "hljs-quote": Object {
          "color": "#408080",
          "fontStyle": "italic",
        },
        "hljs-regexp": Object {
          "color": "#b68",
        },
        "hljs-section": Object {
          "color": "#19469d",
        },
        "hljs-selector-class": Object {
          "color": "#19469d",
        },
        "hljs-selector-id": Object {
          "color": "#19469d",
        },
        "hljs-selector-tag": Object {
          "color": "#954121",
        },
        "hljs-string": Object {
          "color": "#219161",
        },
        "hljs-strong": Object {
          "fontWeight": "bold",
        },
        "hljs-subst": Object {
          "color": "#954121",
        },
        "hljs-symbol": Object {
          "color": "#990073",
        },
        "hljs-tag": Object {
          "color": "#000080",
          "fontWeight": "normal",
        },
        "hljs-template-variable": Object {
          "color": "#008080",
        },
        "hljs-title": Object {
          "color": "#458",
          "fontWeight": "bold",
        },
        "hljs-type": Object {
          "color": "#19469d",
        },
        "hljs-variable": Object {
          "color": "#008080",
        },
      }
    }
  >
    :root {
  --base-font-size: 16px;
  --scale-ratio: 1.414;

  --font-xs:  calc(var(--base-font-size) / var(--scale-ratio));
  --font-sm:  var(--base-font-size);
  --font-md:  calc(var(--font-sm) * var(--scale-ratio));
  --font-lg:  calc(var(--font-md) * var(--scale-ratio));
  --font-xl:  calc(var(--font-lg) * var(--scale-ratio));
  --font-xxl: calc(var(--font-xl) * var(--scale-ratio));

  font-size: var(--font-sm);
}

body {
  background-color: #fff;
  color: #333;
  font-family: 'sans-serif';
  font-weight: 400;
  line-height: 1.5;
}

p {
  margin-bottom: 1.25em;
}

h1, h2, h3, h4, h5 {
  margin: 2.75rem 0 1rem;
  line-height: 1.2;
}

h1 {
  font-size: var(--font-xxl);
  margin-top: 0;
}

h2 {
  font-size: var(--font-xl);
}

h3 {
  font-size: var(--font-lg);
}

h4 {
  font-size: var(--font-md);
}

small {
  font-size: var(--font-xs);
}

  </SyntaxHighlighter>
  <div
    className="controls"
  >
    <Connect(CopyToClipboardButton)
      target="code-snippet"
    />
  </div>
</div>
`;
